apps/studio/components/interfaces/ExplainVisualizer/ExplainVisualizer.parser.ts

```diff
@@ -0,0 +1,256 @@
+    if (detailPatterns.test(content) && stack.length > 0) {
+      // Attach to the most recent node at or above this indentation
+      const currentNode = stack[stack.length - 1].node
+      currentNode.details += (currentNode.details ? '\n' : '') + content
+      continue
+    }
+
+    // Check if this is a continuation of details (indented text without operation pattern)
+    // These are typically wrapped condition expressions
+    if (!hasArrow && stack.length > 0 && leadingSpaces > 0) {
+      const lastItem = stack[stack.length - 1]
+      // If it's more indented than the last node and doesn't look like an operation
+      if (leadingSpaces > lastItem.indent && !content.match(/^\w+.*\(cost=/)) {
+        lastItem.node.details += (lastItem.node.details ? '\n' : '') + content
+        continue
+      }
+    }
+
+    // Parse main operation line: "Operation on table (metrics)"
+    // Match operation with optional metrics in parentheses
+    // Handle multiple metric groups like (cost=...) (actual time=...)
+    const metricsMatch = content.match(/^(.+?)\s*(\([^)]*cost=[^)]+\)(?:\s*\([^)]+\))*)?\s*$/)
+
+    if (!metricsMatch) {
+      continue
+    }
+
+    const [, operationPart, metricsStr] = metricsMatch
+    const metrics = metricsStr
+      ? metricsStr.replace(/^\(|\)$/g, '').replace(/\)\s*\(/g, ' ')
+      : undefined
+
+    // Split operation and object name (e.g., "Seq Scan on users" -> operation: "Seq Scan", details: "users")
+    let operation = operationPart.trim()
+    let details = ''
+
+    // Check for "on tablename" or "using indexname" patterns
+    const onMatch = operationPart.match(/^(.+?)\s+on\s+(.+)$/i)
+    const usingMatch = operationPart.match(/^(.+?)\s+using\s+(.+)$/i)
+
+    if (onMatch) {
+      operation = onMatch[1].trim()
+      details = 'on ' + onMatch[2].trim()
+    } else if (usingMatch) {
+      operation = usingMatch[1].trim()
+      details = 'using ' + usingMatch[2].trim()
+    }
+
+    // Calculate the tree level based on indentation
+    // PostgreSQL typically uses 6 spaces per level for -> nodes
+    const level = hasArrow ? Math.floor(effectiveIndent / 6) + 1 : 0
+
+    const node = createNode(operation, details, metrics, level, line)
+    addNodeToTree(node, effectiveIndent, root, stack)
+  }
+
+  return root
+}
+
+function createNode(
+  operation: string,
+  details: string | undefined,
+  metrics: string | undefined,
+  level: number,
+  raw: string
+): ExplainNode {
+  const node: ExplainNode = {
+    operation: operation.trim(),
+    details: details?.trim() || '',
+    cost: null,
+    rows: null,
+    width: null,
+    actualTime: null,
+    actualRows: null,
+    level,
+    children: [],
+    raw,
+  }
+
+  if (metrics) {
+    // Parse cost=start..end
+    const costMatch = metrics.match(/cost=([\d.]+)\.\.([\d.]+)/)
+    if (costMatch) {
+      node.cost = { start: parseFloat(costMatch[1]), end: parseFloat(costMatch[2]) }
```

Comment:
should NaN checks be done here (for all metrics)? instead of moving defensiveness further down, we can guarantee that everything is either a number or undefined